DSESwift/DatabaseHandler.swift

```diff
@@ -465,8 +466,15 @@ class DatabaseHandler: NSObject{
         return (startTime != nil) && (endTime != nil) && (startTime?.timeIntervalSince1970 >= endTime?.timeIntervalSince1970)
     }
     
-    private static func getUserId() -> String {
-        return KeychainWrapper.stringForKey(KEYCHAIN_USERID)!
+    private static func getUserId() throws -> String {
+        let defaults = NSUserDefaults.standardUserDefaults()
```

Comment:
Because KeyChain has the annoying limitation that does not return the stored values until the moment that the user unlock the screen for the first time counting from a reboot. userId is required to store data into DSE, which could happen before the first unlock. This will result in losing data points. Plus UserId is not really privacy sensitive info, so we use UserDefaults for userId.